internal/api/impl/proxy/proxy.go

```diff
@@ -424,15 +424,20 @@ func (h *httpProxy) serve(c echo.Context) error {
 	// Reverse proxy request.
 	reverseProxy.ServeHTTP(res, req)
 	// Return any error handled during proxying request.
-	return proxyErr
+	if proxyErr != nil {
+		// we need to wrap the error with an Echo Error,
+		// otherwise the error will be hidden by the middleware "middleware.HandleError".
+		return echo.NewHTTPError(res.Status, proxyErr.Error())
```

Comment:
this is the actual fix, the rest of the changes is just an alignment about what kind of error an endpoint should use, and it should use the error `shared.PersesError` if it's possible. And as you can see, that's not really possible for this fix ;)